subprojects/docs/src/docs/userguide/gradleWrapper.adoc

```diff
@@ -151,20 +151,20 @@ This can be used in conjunction with a proxy, authenticated or not. See <<sec:ac
 
 The Gradle Wrapper allows for verification of the downloaded Gradle distribution via SHA-256 hash sum comparison. This increases security against targeted attacks by preventing a man-in-the-middle attacker from tampering with the downloaded Gradle distribution.
 
-To enable this feature you'll want to first calculate the SHA-256 hash of a known Gradle distribution. You can generate a SHA-256 hash from Linux and OSX or Windows (via https://www.cygwin.com/[Cygwin]) with the `shasum` command.
+To enable this feature, download the `.sha256` file associated with the Gradle distribution you want to verify.
```

Comment:
:o: This would be even clearer if we gave a specific link to the currently selected Gradle version (which I think should be possible, this would only be in the Gradle 4.3+ docs, no backfill).

Reply:
@eriwen, I don't really understand exactly what you're suggesting that we link to?

1. Link to https://gradle.org/install/#manually section that has links for the `-bin.zip`, `-bin.zip.sha256`, `-all.zip`, `-all.zip.sha256`
1. Actually inline links to each of those four files directly from the user guide?
1. Only link to the `-bin.zip.sha256` file to just show the user what one looks like?

I feel like it must not be the first option, because I've already linked to that page. But, I'm not really sure which of the other options makes sense for the user guide. Here is what I'm worried might happen with each of them.

For option 2: Someone who ends up looking at an older version of the user guide might not realize that we're linking to the download links for an "out-of-date" version of Gradle, and download the distribution from there. Also, if this becomes a popular place to find the versions, we could start messing with our analytics.

For option 3: I worry that providing a link to the `.sha256` without the corresponding `.zip` file may cause some confusion. Suppose your wrapper points to some other version of Gradle, and you come here to figure out how to download and verify a `.sha256`.  You download it, and then try to add it to your `wrapper/gradle-wrapper.properties` and find that the verification fails.

I did consider including a screenshot of what the download links look like from one of the pages. (Maybe even with a green arrow pointing to it.) Would that solve the same problem you're getting at?

Reply:
So, I've looked into this a little bit more, and I cannot find an example in our `.adoc` files which make up the userguide of embedding the version of Gradle into the userguide. I was going to suggest something like the following text:

For example, if you wanted to verify the [current binary-only distribution](https://services.gradle.org/distributions/gradle-4.1-bin.zip)(version 4.1) of Gradle, you could use the corresponding [sha256 checksum file](https://services.gradle.org/distributions/gradle-4.1-bin.zip.sha256) with the instructions below.

But, to do that at generation time, would require that we can know the current version of the userguide when we're generating the asciidoc. I'm not aware of such a mechanism (although, it may be there.)

Reply:
Okay, since this original comment was optional, I'm going to go ahead and merge this, and update the associated forum threads. If we want to make it clearer in the future, we can create another pull request.